second commit

dimyalt opened the pull request.

dimyalt committed:
second commit

mkouznetsov-git commented on src/main/java/sitepages/data/DriverOptions.java:
```diff
@@ -0,0 +1,33 @@
+
+
+public class DriverOptions {
+    private final String CHROME_PATH = "C:\\Program Files\\Google\\Chrome\\Application\\chrome.exe";
```
Далеко не у всех на компьютерах драйвера лежат именно в этих папках. Даже винда не у всех стоит. Лучше использовать дефолтный драйвер, тот, который прописан в системе, без указания пути размещения. В крайнем случае можно в readme указать ссылку где эти драйвера можно скачать.

mkouznetsov-git commented on src/test/java/TestLoginSuccessfull.java:
```diff
@@ -0,0 +1,71 @@
+    @Test
+    @DisplayName("Логин пользователя")
+    public void userLogin() throws Exception {
+        if(browser.equals("chrome")){
```
Выбор драйвера и все подготовительные процедуры надо вынести в метод с аннотацией `Before`.
У тебя эта конструкция повторяется в других тестовых классах тоже -- тогда лучше сделать базовый тестовый класс, там сделать метод с аннотацией `Before`, перенести в него всю эту логику и наследовать тестовые классы от этого базового класса. Это сильно сократит лишний код в самих тестовых классах и упростит проект.

mkouznetsov-git commented on src/test/java/TestLoginSuccessfull.java:
```diff
@@ -0,0 +1,71 @@
+        }
+        driver.get(PAGE_URL); //Переходим на главную страницу
+        LoginStellarburgers objLoginPage = new LoginStellarburgers(driver); //Создаем объект класса страницы логина
+        assertTrue(objLoginPage.getUserLogin(pointOfEntry, userEmail, userPassword));
```
Вся логика проверки спрятана в других классах, это неправильно. Тестовый класс для того и тестовый, чтобы в нём была описана вся логика тестирования. Надо перенести логику проверки сюда. Вспомогательные классы -- для каких-то общих блоков кода, помогающих сократить лишний код в тестах, но не для того, чтобы раскладывать по ним саму стратегию тестирования.

mkouznetsov-git commented on src/test/java/TestLogoutSuccesfull.java:
```diff
@@ -0,0 +1,49 @@
+        assertTrue(objAccountPage.getLogoutStatus());
+    }
+    @Test
+    @DisplayName("Выход из личного кабинета в Chrome")
```
Хром или Яндекс?

mkouznetsov-git commented on src/test/java/TestLogoutSuccesfull.java:
```diff
@@ -0,0 +1,49 @@
+
+    @Test
+    @DisplayName("Выход из личного кабинета в Chrome")
+    public void logoutUserAccountChrome() throws Exception {
```
Два одинаковых тестовых метода -- это лишнее дублирование кода. Параметризуй то, что изменяется и оставь один тестовый метод.

mkouznetsov-git commented on src/main/java/sitepages/AccountStellarburgers.java:
```diff
@@ -0,0 +1,64 @@
+        driver.findElement(burgerLoginButton).click(); // Кликнули по кнопке "Войти в аккаунт"
+        new WebDriverWait(driver, Duration.ofSeconds(2)) // Ждем 2 секунды до появления поля ввода Email
+                .until(ExpectedConditions.visibilityOfElementLocated(fieldEmailLogin));
+        driver.findElement(fieldEmailLogin).isEnabled();
```
Этот метод возвращает true или false, но с этими значениями ты ничего не делаешь. В текущем виде эта строка лишняя.

mkouznetsov-git commented on src/main/java/sitepages/AccountStellarburgers.java:
```diff
@@ -0,0 +1,64 @@
+        driver.findElement(fieldEmailLogin).isEnabled();
+        driver.findElement(fieldEmailLogin).clear();
+        driver.findElement(fieldEmailLogin).sendKeys(setUserEmail);
+        driver.findElement(fieldPasswordLogin).isEnabled();
```
То же самое

mkouznetsov-git commented on src/main/java/sitepages/AccountStellarburgers.java:
```diff
@@ -0,0 +1,64 @@
+    public AccountStellarburgers(WebDriver driver) {
+        this.driver = driver;
+    }
+    @Step("Логинимся и заходим в личный кабинет")
```
Один метод должен содержать одно логическое действие. В этом методе написано 2 действия. Их надо разделить на два метода: первый метод -- логин, второй -- вход в ЛК.

mkouznetsov-git commented on src/test/java/TestLoginSuccessfull.java:
```diff
@@ -0,0 +1,71 @@
+@RunWith(Parameterized.class)
+public class TestLoginSuccessfull {
+    WebDriver driver;
+    private static final String PAGE_URL = "https://stellarburgers.nomoreparties.site";
```
PAGE_URL, пользователя и пароль лучше вынести в базовый класс, они у тебя много где повторяются.

mkouznetsov-git commented on src/main/java/sitepages/UserPersonalAccountStarburgers.java:
```diff
@@ -0,0 +1,59 @@
+        driver.findElement(burgerLoginButton).click(); // Кликнули по кнопке "Войти в аккаунт"
+        new WebDriverWait(driver, Duration.ofSeconds(2)) // Ждем 2 секунды до появления поля ввода Email
+                .until(ExpectedConditions.visibilityOfElementLocated(fieldEmailLogin));
+        driver.findElement(fieldEmailLogin).isEnabled();
```
Эта строчка кода по сути ничего не делает -- она лишняя

mkouznetsov-git commented on src/main/java/sitepages/UserPersonalAccountStarburgers.java:
```diff
@@ -0,0 +1,59 @@
+        driver.findElement(fieldEmailLogin).isEnabled();
+        driver.findElement(fieldEmailLogin).clear();
+        driver.findElement(fieldEmailLogin).sendKeys(setUserEmail);
+        driver.findElement(fieldPasswordLogin).isEnabled();
```
Эта строчка кода по сути ничего не делает -- она лишняя. Посмотри в других классах на предмет подобных кейсов и удали лишнее, если не нужно.

dimyalt committed:
1. Tried to remove repetitions
2. Removed methods and lines that do nothing
3. Used the default driver for Chrome
4. Moved duplicate lines to the base class
5. Moved preparatory procedures to @Before
6. Moved the logic of checks to test classes

mkouznetsov-git commented on src/test/java/TestRegistrationWithGoodPasswordOk.java:
```diff
@@ -0,0 +1,68 @@
+        driver.findElement(RegisterStellarburgers.getButtonRegistration()).click();
+        new WebDriverWait(driver, Duration.ofSeconds(2))
+                .until(ExpectedConditions.visibilityOfElementLocated(RegisterStellarburgers.getButtonEnter()));
+        assertEquals(RegisterStellarburgers.getExpectedURL(), driver.getCurrentUrl());
```
Вообще текущий URL в адресной строке не может быть гарантией того, что регистрация прошла успешно. Надо зацепиться за что-то более очевидное. )

mkouznetsov-git commented on src/test/java/TestLoginSuccessful.java:
```diff
@@ -0,0 +1,94 @@
+    @DisplayName("Логин пользователя")
+    public void userLogin(){
+        switch (pointOfEntry) {
+            case "Войти в аккаунт":
```
Эти кейсы лучше разбить по отдельным методам, а то, что общее внизу -- вынести в отдельный локальный метод и вызывать из этих тестовых методов.

mkouznetsov-git commented on src/test/java/TestSwitchingBetweenSectionsOfTheConstructor.java:
```diff
@@ -0,0 +1,81 @@
+    public void transitionConstructorSectors(){
+        ConstructorStellarburgers objConstructorPage = new ConstructorStellarburgers(); //Создаем объект класса страницы конструктора
+        driver.get(objConstructorPage.getMainPageTestURL()); //Переходим на главную страницу
+        switch (sector) {
```
Здесь тоже лучше разбить на методы. По сути у тебя 3 тестовых метода для трёх разделов. И параметров тогда будет в 3 раза меньше )

dimyalt committed:
1. Removed the "current page address" from the check
2. Reworked cases
